src/error.rs

```diff
@@ -1,31 +1,82 @@
+use std::error::Error;
 use std::ffi::{c_int, c_uint};
+use std::fmt::{self, Display, Formatter};
 
 use strum::FromRepr;
 
+/// Error enum return by various `rav1d` operations.
 #[derive(Clone, Copy, PartialEq, Eq, FromRepr, Debug)]
 #[repr(u8)]
 #[non_exhaustive]
 pub enum Rav1dError {
     /// This represents a generic `rav1d` error.
-    /// It has nothing to do with the other `errno`-based ones
-    /// (and that's why it's not all caps like the other ones).
+    /// It has nothing to do with the other `errno`-based ones.
     ///
     /// Normally `EPERM = 1`, but `dav1d` never uses `EPERM`,
     /// but does use `-1`, as opposed to the normal `DAV1D_ERR(E*)`.
     ///
-    /// Also Note that this forces `0` to be the niche,
+    /// Also note that this forces `0` to be the niche,
     /// which is more optimal since `0` is no error for [`Dav1dResult`].
-    EGeneric = 1,
-
-    ENOENT = libc::ENOENT as u8,
-    EIO = libc::EIO as u8,
-    EAGAIN = libc::EAGAIN as u8,
-    ENOMEM = libc::ENOMEM as u8,
-    EINVAL = libc::EINVAL as u8,
-    ERANGE = libc::ERANGE as u8,
-    ENOPROTOOPT = libc::ENOPROTOOPT as u8,
+    Other = 1,
+
+    /// No sequence header.
+    ///
+    /// No Sequence Header OBUs were found in the buffer.
+    NoSequenceHeader = libc::ENOENT as u8,
```

Comment:
I'd rather this go back to just `NoEntity`. Since this will now be part of the public API, if `dav1d` changes to return some other `ENOENT` error somewhere, we'll have to rename this and cause a breaking change.

We can keep the more specific docs, though, like you're doing for `OutOfRange` and the frame size.